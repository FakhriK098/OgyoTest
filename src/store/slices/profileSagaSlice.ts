import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { IProfile } from 'src/types/profile';

interface IProfileState {
  profile: IProfile | null;
  loading: boolean;
  error: string | null;
  localAvatarUri: string | null;
}

const initialState: IProfileState = {
  profile: null,
  loading: false,
  error: null,
  localAvatarUri: null,
};

export const profileSagaSlice = createSlice({
  name: 'profile',
  initialState,
  reducers: {
    fetchProfileRequest: state => {
      state.loading = true;
      state.error = null;
    },
    fetchProfileSuccess: (state, action: PayloadAction<IProfile>) => {
      state.loading = false;
      state.profile = action.payload;
      state.error = null;
    },
    fetchProfileFailure: (state, action: PayloadAction<string>) => {
      state.loading = false;
      state.error = action.payload;
    },
    setLocalAvatarUri: (state, action: PayloadAction<string | null>) => {
      state.localAvatarUri = action.payload;
    },
  },
});

export const { fetchProfileRequest, fetchProfileSuccess, fetchProfileFailure, setLocalAvatarUri } =
  profileSagaSlice.actions;

export default profileSagaSlice.reducer;
