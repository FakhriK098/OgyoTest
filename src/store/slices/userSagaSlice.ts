import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { IUser } from 'src/types/user';

interface IUserState {
  user: IUser | null;
  loading: boolean;
  error: string | null;
}

const initialState: IUserState = {
  user: null,
  loading: false,
  error: null,
};

export const userSagaSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    fetchUserRequest: (
      state,
      _action: PayloadAction<{ fullName?: string }>,
    ) => {
      state.loading = true;
      state.error = null;
    },
    fetchUserSuccess: (state, action: PayloadAction<IUser>) => {
      state.loading = false;
      state.user = action.payload;
      state.error = null;
    },
    fetchUserFailure: (state, action: PayloadAction<string>) => {
      state.loading = false;
      state.error = action.payload;
    },
  },
});

export const { fetchUserRequest, fetchUserSuccess, fetchUserFailure } =
  userSagaSlice.actions;

export default userSagaSlice.reducer;
