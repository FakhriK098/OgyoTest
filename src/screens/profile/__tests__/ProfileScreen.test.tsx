import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import ProfileScreen from '../ProfileScreen';
import { profileSagaSlice, setLocalAvatarUri } from 'src/store/slices/profileSagaSlice';
import * as ImagePicker from 'react-native-image-picker';
import { Alert } from 'react-native';
import * as reduxHooks from 'src/hooks/redux';

jest.mock('src/hooks/redux', () => ({
  useAppDispatch: jest.fn(),
  useAppSelector: jest.fn(),
}));

const mockProfile = {
  login: 'testuser',
  id: 1,
  node_id: 'test-node',
  avatar_url: 'https://example.com/avatar.jpg',
  gravatar_id: '',
  url: 'https://api.github.com/users/testuser',
  html_url: 'https://github.com/testuser',
  followers_url: 'https://api.github.com/users/testuser/followers',
  following_url: 'https://api.github.com/users/testuser/following{/other_user}',
  gists_url: 'https://api.github.com/users/testuser/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/testuser/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/testuser/subscriptions',
  organizations_url: 'https://api.github.com/users/testuser/orgs',
  repos_url: 'https://api.github.com/users/testuser/repos',
  events_url: 'https://api.github.com/users/testuser/events{/privacy}',
  received_events_url: 'https://api.github.com/users/testuser/received_events',
  type: 'User',
  user_view_type: 'public',
  site_admin: false,
  name: 'Test User',
  company: null,
  blog: '',
  location: null,
  email: null,
  hireable: null,
  bio: null,
  twitter_username: null,
  public_repos: 10,
  public_gists: 0,
  followers: 100,
  following: 50,
  created_at: '2020-01-01T00:00:00Z',
  updated_at: '2023-01-01T00:00:00Z',
};

const createMockStore = (initialState = {}) => {
  return configureStore({
    reducer: {
      profile: profileSagaSlice.reducer,
    },
    preloadedState: {
      profile: {
        profile: mockProfile,
        loading: false,
        error: null,
        localAvatarUri: null,
        ...initialState,
      },
    },
  });
};

describe('ProfileScreen', () => {
  const mockDispatch = jest.fn();
  
  beforeEach(() => {
    jest.clearAllMocks();
    (reduxHooks.useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);
  });

  it('should render loading state', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: true,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore({ loading: true });
    const { getByTestId } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(() => getByTestId('activity-indicator')).not.toThrow();
  });

  it('should render profile data correctly', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore();
    const { getByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(getByText('testuser')).toBeTruthy();
    expect(getByText('100')).toBeTruthy(); // followers
    expect(getByText('50')).toBeTruthy(); // following
    expect(getByText('10')).toBeTruthy(); // repositories
  });

  it('should render empty when profile is null', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: null,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore({ profile: null });
    const { toJSON } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(toJSON()).toBeNull();
  });

  it('should render empty when there is an error', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: 'Failed to fetch profile',
      localAvatarUri: null,
    });
    
    const store = createMockStore({ error: 'Failed to fetch profile' });
    const { toJSON } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(toJSON()).toBeNull();
  });

  it('should open modal when camera button is pressed', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore();
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    expect(getByText('Choose Photo')).toBeTruthy();
    expect(getByText('Take Photo')).toBeTruthy();
    expect(getByText('Choose from Gallery')).toBeTruthy();
  });

  it('should close modal when cancel is pressed', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore();
    const { getByTestId, getByText, queryByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    const cancelButton = getByText('Cancel');
    fireEvent.press(cancelButton);

    expect(queryByText('Choose Photo')).toBeNull();
  });

  it('should handle image selection from gallery', async () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const mockImageResponse = {
      assets: [{ uri: 'file://new-image.jpg' }],
    };
    
    (ImagePicker.launchImageLibrary as jest.Mock).mockImplementation((_options, callback) => {
      callback(mockImageResponse);
    });

    const store = createMockStore();
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    const galleryButton = getByText('Choose from Gallery');
    fireEvent.press(galleryButton);

    await waitFor(() => {
      expect(ImagePicker.launchImageLibrary).toHaveBeenCalled();
      expect(mockDispatch).toHaveBeenCalledWith(setLocalAvatarUri('file://new-image.jpg'));
    });
  });

  it('should handle camera capture', async () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const mockImageResponse = {
      assets: [{ uri: 'file://camera-image.jpg' }],
    };
    
    (ImagePicker.launchCamera as jest.Mock).mockImplementation((_options, callback) => {
      callback(mockImageResponse);
    });

    const store = createMockStore();
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    const takePhotoButton = getByText('Take Photo');
    fireEvent.press(takePhotoButton);

    await waitFor(() => {
      expect(ImagePicker.launchCamera).toHaveBeenCalled();
      expect(mockDispatch).toHaveBeenCalledWith(setLocalAvatarUri('file://camera-image.jpg'));
    });
  });

  it('should handle image picker cancellation', async () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const mockImageResponse = {
      didCancel: true,
    };
    
    (ImagePicker.launchImageLibrary as jest.Mock).mockImplementation((_options, callback) => {
      callback(mockImageResponse);
    });

    const store = createMockStore();
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    const galleryButton = getByText('Choose from Gallery');
    fireEvent.press(galleryButton);

    await waitFor(() => {
      expect(ImagePicker.launchImageLibrary).toHaveBeenCalled();
      expect(mockDispatch).not.toHaveBeenCalledWith(setLocalAvatarUri(expect.anything()));
    });
  });

  it('should handle image picker error', async () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const mockImageResponse = {
      errorMessage: 'Permission denied',
    };
    
    jest.spyOn(Alert, 'alert');
    
    (ImagePicker.launchImageLibrary as jest.Mock).mockImplementation((_options, callback) => {
      callback(mockImageResponse);
    });

    const store = createMockStore();
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    const galleryButton = getByText('Choose from Gallery');
    fireEvent.press(galleryButton);

    await waitFor(() => {
      expect(Alert.alert).toHaveBeenCalledWith('Error', 'Permission denied');
    });
  });

  it('should display local avatar when set', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: 'file://local-avatar.jpg',
    });
    
    const store = createMockStore({ localAvatarUri: 'file://local-avatar.jpg' });
    const { getByTestId } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const avatar = getByTestId('avatar-image');
    expect(avatar.props.source.uri).toBe('file://local-avatar.jpg');
  });

  it('should fallback to profile avatar when local avatar is not set', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore();
    const { getByTestId } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const avatar = getByTestId('avatar-image');
    expect(avatar.props.source.uri).toBe('https://example.com/avatar.jpg');
  });

  it('should match snapshot with profile data', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore();
    const { toJSON } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot in loading state', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: true,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore({ loading: true });
    const { toJSON } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with local avatar', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: 'file://local-avatar.jpg',
    });
    
    const store = createMockStore({ localAvatarUri: 'file://local-avatar.jpg' });
    const { toJSON } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with modal open', () => {
    (reduxHooks.useAppSelector as jest.Mock).mockReturnValue({
      profile: mockProfile,
      loading: false,
      error: null,
      localAvatarUri: null,
    });
    
    const store = createMockStore();
    const { getByTestId, toJSON } = render(
      <Provider store={store}>
        <ProfileScreen />
      </Provider>
    );

    const cameraButton = getByTestId('camera-button');
    fireEvent.press(cameraButton);

    expect(toJSON()).toMatchSnapshot();
  });
});