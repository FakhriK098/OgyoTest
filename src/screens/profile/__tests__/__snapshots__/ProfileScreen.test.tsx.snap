// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen should match snapshot in loading state 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#161B28"
      size="large"
      testID="activity-indicator"
    />
  </View>
</RCTSafeAreaView>
`;

exports[`ProfileScreen should match snapshot with local avatar 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="camera-button"
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 50,
              "elevation": 8,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
            }
          }
        >
          <Image
            source={
              {
                "uri": "file://local-avatar.jpg",
              }
            }
            style={
              [
                {
                  "borderRadius": 50,
                },
                {
                  "height": 100,
                  "width": 100,
                },
              ]
            }
            testID="avatar-image"
          />
        </View>
        <View
          style={
            {
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/camera.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 8,
          "marginHorizontal": 8,
          "marginTop": 16,
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4.65,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "600",
            "marginBottom": 16,
            "textAlign": "center",
          }
        }
      >
        testuser
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Followers:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          100
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Following:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          50
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Repositories:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          10
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`ProfileScreen should match snapshot with modal open 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="camera-button"
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 50,
              "elevation": 8,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://example.com/avatar.jpg",
              }
            }
            style={
              [
                {
                  "borderRadius": 50,
                },
                {
                  "height": 100,
                  "width": 100,
                },
              ]
            }
            testID="avatar-image"
          />
        </View>
        <View
          style={
            {
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/camera.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 8,
          "marginHorizontal": 8,
          "marginTop": 16,
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4.65,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "600",
            "marginBottom": 16,
            "textAlign": "center",
          }
        }
      >
        testuser
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Followers:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          100
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Following:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          50
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Repositories:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          10
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 12,
            "elevation": 5,
            "maxWidth": 320,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#000000",
              "fontSize": 18,
              "fontWeight": "600",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Choose Photo
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#EEEBE3",
              "borderRadius": 8,
              "marginBottom": 10,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 14,
            }
          }
        >
          <Text
            style={
              {
                "color": "#161B28",
                "fontSize": 16,
              }
            }
          >
            Take Photo
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#EEEBE3",
              "borderRadius": 8,
              "marginBottom": 10,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 14,
            }
          }
        >
          <Text
            style={
              {
                "color": "#161B28",
                "fontSize": 16,
              }
            }
          >
            Choose from Gallery
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#DEDEDE",
              "borderRadius": 8,
              "marginBottom": 10,
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 14,
            }
          }
        >
          <Text
            style={
              {
                "color": "#161B28",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</RCTSafeAreaView>
`;

exports[`ProfileScreen should match snapshot with profile data 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EEEBE3",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "justifyContent": "center",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="camera-button"
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 50,
              "elevation": 8,
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://example.com/avatar.jpg",
              }
            }
            style={
              [
                {
                  "borderRadius": 50,
                },
                {
                  "height": 100,
                  "width": 100,
                },
              ]
            }
            testID="avatar-image"
          />
        </View>
        <View
          style={
            {
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/camera.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "elevation": 8,
          "marginHorizontal": 8,
          "marginTop": 16,
          "paddingBottom": 8,
          "paddingHorizontal": 16,
          "paddingTop": 16,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4.65,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "600",
            "marginBottom": 16,
            "textAlign": "center",
          }
        }
      >
        testuser
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Followers:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          100
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Following:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          50
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "600",
              "marginRight": 4,
            }
          }
        >
          Repositories:
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "400",
              "marginRight": 4,
            }
          }
        >
          10
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
